Pylint: Fix `unspecified-encoding` issues.

## 💡 Description
Update the codebase to use "utf-8" encoding for text-based file access. `pylint` report is attached (`pylint . --disable=all --enable=unspecified-encoding`) (71 occurrences):
```
************* Module pds.naif_pds4_bundler.pipeline.runtime
pds/naif_pds4_bundler/pipeline/runtime.py:27:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
************* Module pds.naif_pds4_bundler.classes.bundle
pds/naif_pds4_bundler/classes/bundle.py:744:17: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
************* Module pds.naif_pds4_bundler.classes.plan
pds/naif_pds4_bundler/classes/plan.py:45:17: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/plan.py:54:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
************* Module pds.naif_pds4_bundler.classes.list
pds/naif_pds4_bundler/classes/list.py:114:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/list.py:349:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/list.py:376:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/list.py:380:21: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/list.py:427:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/list.py:593:21: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/list.py:650:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/list.py:736:21: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
************* Module pds.naif_pds4_bundler.classes.setup
pds/naif_pds4_bundler/classes/setup.py:68:17: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/setup.py:747:21: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/setup.py:1148:17: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/setup.py:1162:17: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/setup.py:1226:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
************* Module pds.naif_pds4_bundler.classes.collection.collection_kernels
pds/naif_pds4_bundler/classes/collection/collection_kernels.py:256:17: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/collection/collection_kernels.py:447:17: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
************* Module pds.naif_pds4_bundler.classes.label.pds3_spice_kernel
pds/naif_pds4_bundler/classes/label/pds3_spice_kernel.py:161:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/label/pds3_spice_kernel.py:164:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/label/pds3_spice_kernel.py:167:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/label/pds3_spice_kernel.py:237:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
************* Module pds.naif_pds4_bundler.classes.label.pds4_inventory
pds/naif_pds4_bundler/classes/label/pds4_inventory.py:64:12: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
************* Module pds.naif_pds4_bundler.classes.label.label
pds/naif_pds4_bundler/classes/label/label.py:406:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/label/label.py:407:17: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
************* Module pds.naif_pds4_bundler.classes.product.product_metakernel
pds/naif_pds4_bundler/classes/product/product_metakernel.py:707:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/product/product_metakernel.py:708:17: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
************* Module pds.naif_pds4_bundler.classes.product.product_inventory
pds/naif_pds4_bundler/classes/product/product_inventory.py:164:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/product/product_inventory.py:166:21: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/product/product_inventory.py:208:17: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/product/product_inventory.py:263:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/product/product_inventory.py:340:21: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
************* Module pds.naif_pds4_bundler.classes.product.product_orbnum
pds/naif_pds4_bundler/classes/product/product_orbnum.py:237:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/product/product_orbnum.py:276:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/product/product_orbnum.py:298:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/product/product_orbnum.py:360:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/product/product_orbnum.py:463:17: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/product/product_orbnum.py:464:21: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
************* Module pds.naif_pds4_bundler.classes.product.product_kernel
pds/naif_pds4_bundler/classes/product/product_kernel.py:204:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/product/product_kernel.py:238:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/product/product_kernel.py:264:17: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/product/product_kernel.py:343:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
************* Module pds.naif_pds4_bundler.classes.product.product_readme
pds/naif_pds4_bundler/classes/product/product_readme.py:78:17: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/product/product_readme.py:79:21: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
************* Module pds.naif_pds4_bundler.classes.product.product_checksum
pds/naif_pds4_bundler/classes/product/product_checksum.py:204:17: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/product/product_checksum.py:510:25: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/product/product_checksum.py:547:17: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
************* Module pds.naif_pds4_bundler.classes.product.product_spiceds
pds/naif_pds4_bundler/classes/product/product_spiceds.py:151:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/product/product_spiceds.py:152:17: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/product/product_spiceds.py:180:17: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/classes/product/product_spiceds.py:182:17: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
************* Module pds.naif_pds4_bundler.utils.time
pds/naif_pds4_bundler/utils/time.py:399:9: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
************* Module pds.naif_pds4_bundler.utils.files
pds/naif_pds4_bundler/utils/files.py:245:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/utils/files.py:246:17: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/utils/files.py:281:9: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/utils/files.py:283:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/utils/files.py:312:9: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/utils/files.py:387:9: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/utils/files.py:481:17: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/utils/files.py:579:9: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/utils/files.py:581:9: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/utils/files.py:605:20: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/utils/files.py:738:17: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/utils/files.py:763:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/utils/files.py:833:9: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/utils/files.py:855:19: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/utils/files.py:863:19: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/utils/files.py:905:9: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/utils/files.py:1069:9: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
pds/naif_pds4_bundler/utils/files.py:1126:9: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
```

--- a/src/pds/naif_pds4_bundler/classes/bundle.py
+++ b/src/pds/naif_pds4_bundler/classes/bundle.py
@@ -741,7 +741,7 @@ class Bundle:
                 f"/miscellaneous"
                 f"/checksum/checksum_v{rel:03d}.tab"
             )
-            with open(checksum_file) as c:
+            with open(checksum_file, encoding='utf-8') as c:
                 for line in c:
                     #
                     # Need to convert the file names to lower case because this

--- a/src/pds/naif_pds4_bundler/classes/collection/collection_kernels.py
+++ b/src/pds/naif_pds4_bundler/classes/collection/collection_kernels.py
@@ -253,7 +253,7 @@ class SpiceKernelsCollection(Collection):
             )
             bundles.sort()
 
-            with open(bundles[-1], "r") as b:
+            with open(bundles[-1], "r", encoding='utf-8') as b:
                 for line in b:
                     if "<start_date_time>" in line:
                         prev_increment_start = line.split(">")[-2].split("<")[0]
@@ -444,7 +444,7 @@ class SpiceKernelsCollection(Collection):
         products = self.product
         for product in products:
             label_name = product.label.name
-            with open(label_name, "r") as p:
+            with open(label_name, "r", encoding='utf-8') as p:
                 for line in p:
                     for element in elements:
                         if element in line:

--- a/src/pds/naif_pds4_bundler/classes/label/label.py
+++ b/src/pds/naif_pds4_bundler/classes/label/label.py
@@ -403,8 +403,8 @@ class PDSLabel:
         if "inventory" in label_name:
             label_name = label_name.replace("inventory_", "")
 
-        with open(label_name, "w+") as f:
-            with open(self.template, "r") as t:
+        with open(label_name, "w+", encoding='utf-8') as f:
+            with open(self.template, "r", encoding='utf-8') as t:
                 for line in t:
                     line = line.rstrip()
                     for key, value in label_dictionary.items():

--- a/src/pds/naif_pds4_bundler/classes/label/pds3_spice_kernel.py
+++ b/src/pds/naif_pds4_bundler/classes/label/pds3_spice_kernel.py
@@ -158,13 +158,13 @@ class SpiceKernelPDS3Label(PDSLabel):
         kernel architecture specification and removes extra empty lines at the
         end of the kernel file.
         """
-        with open(self.name, "r") as label:
+        with open(self.name, "r", encoding='utf-8') as label:
             label_lines = label.readlines()
 
-        with open(self.product.path, "r+") as kernel:
+        with open(self.product.path, "r+", encoding='utf-8') as kernel:
             kernel_lines = kernel.readlines()
 
-        with open(self.product.path, "w") as kernel:
+        with open(self.product.path, "w", encoding='utf-8') as kernel:
 
             if "KPL/" in kernel_lines[0]:
                 kernel.write(kernel_lines[0])
@@ -234,7 +234,7 @@ class SpiceKernelPDS3Label(PDSLabel):
         The routine inserts the label in the kernel comment.
         """
         label_lines = []
-        with open(self.name, "r") as label:
+        with open(self.name, "r", encoding='utf-8') as label:
             for line in label:
                 if line.strip() != "END":
                     label_lines.append(line.rstrip())

--- a/src/pds/naif_pds4_bundler/classes/label/pds4_inventory.py
+++ b/src/pds/naif_pds4_bundler/classes/label/pds4_inventory.py
@@ -61,7 +61,7 @@ class InventoryPDS4Label(PDSLabel):
         #
         # Count number of lines in the inventory file
         #
-        f = open(self.product.path)
+        f = open(self.product.path, encoding='utf-8')
         self.N_RECORDS = str(len(f.readlines()))
         f.close()
 

--- a/src/pds/naif_pds4_bundler/classes/list.py
+++ b/src/pds/naif_pds4_bundler/classes/list.py
@@ -111,7 +111,7 @@ class KernelList:
             self, self.setup.working_directory + os.sep + list_name, list_dictionary
         )
 
-        with open(self.setup.working_directory + os.sep + list_name, "a+") as f:
+        with open(self.setup.working_directory + os.sep + list_name, "a+", encoding='utf-8') as f:
 
             for kernel in self.kernel_list:
                 ker_added_to_list = False
@@ -346,7 +346,7 @@ class KernelList:
         # Generate the kernel list attribute, necessary for the validation.
         #
         kernels = []
-        with open(kernel_list, "r") as lst:
+        with open(kernel_list, "r", encoding='utf-8') as lst:
             for line in lst:
                 if "FILE             =" in line:
                     kernels.append(line.split(os.sep)[-1][:-1])
@@ -373,11 +373,11 @@ class KernelList:
         complete_list = f"{self.setup.mission_acronym}_complete.kernel_list"
 
         release_list = []
-        with open(self.setup.working_directory + os.sep + complete_list, "w+") as c:
+        with open(self.setup.working_directory + os.sep + complete_list, "w+", encoding='utf-8') as c:
             for kernel_list in kernel_lists:
                 logging.info(f"-- Adding {kernel_list}")
                 release_list.append(int(kernel_list.replace("_", ".").split(".")[-3]))
-                with open(kernel_list, "r") as lst:
+                with open(kernel_list, "r", encoding='utf-8') as lst:
                     for line in lst:
                         c.write(line)
 
@@ -424,7 +424,7 @@ class KernelList:
             for err in errors:
                 logging.error(f"   {err}")
 
-        with open(list_path, "r") as lst:
+        with open(list_path, "r", encoding='utf-8') as lst:
 
             #
             # Check that the list has the same number of FILE,
@@ -590,7 +590,7 @@ class KernelList:
             ker_in_list = []
             for kernel_list in kernel_lists:
 
-                with open(kernel_list, "r") as lst:
+                with open(kernel_list, "r", encoding='utf-8') as lst:
 
                     #
                     # Check that the list has the same number of FILE,
@@ -648,7 +648,7 @@ class KernelList:
         opt_in_list = []
 
         with open(
-            self.setup.working_directory + os.sep + self.complete_list, "r"
+            self.setup.working_directory + os.sep + self.complete_list, "r", encoding='utf-8'
         ) as lst:
 
             #
@@ -733,7 +733,7 @@ class KernelList:
                     self.setup.root_dir + f"/config/{self.setup.mission_acronym}"
                     f"_mission_template.pds"
                 )
-                with open(template, "r") as o:
+                with open(template, "r", encoding='utf-8') as o:
                     template_lines = o.readlines()
 
                 for option in opt_in_list:

--- a/src/pds/naif_pds4_bundler/classes/plan.py
+++ b/src/pds/naif_pds4_bundler/classes/plan.py
@@ -42,7 +42,7 @@ class ReleasePlan:
         # as a release plan. If so, a plan is generated.
         if plan.suffix != ".plan" and self.setup.args.faucet == "labels":
             plan = os.path.join(self.setup.working_directory, self._plan_name())
-            with open(plan, "w") as pl:
+            with open(plan, "w", encoding='utf-8') as pl:
                 pl.write(plan.split(os.sep)[-1])
 
         elif plan.suffix != ".plan":
@@ -51,7 +51,7 @@ class ReleasePlan:
                 "kernels are only allowed in labeling mode."
             )
 
-        with open(plan, "r") as f:
+        with open(plan, "r", encoding='utf-8') as f:
             for line in f:
                 ker_matched = False
                 for pattern in patterns:

--- a/src/pds/naif_pds4_bundler/classes/product/product_checksum.py
+++ b/src/pds/naif_pds4_bundler/classes/product/product_checksum.py
@@ -201,7 +201,7 @@ class ChecksumProduct(Product):
         # the miscellaneous collection was present in the release.
         #
         if self.path_current:
-            with open(self.path_current, "r") as c:
+            with open(self.path_current, "r", encoding='utf-8') as c:
                 for line in c:
                     #
                     # Check the format of the current checksum file.
@@ -507,7 +507,7 @@ class ChecksumProduct(Product):
                         )
 
                 if os.path.isfile(path):
-                    with open(path, "r") as lbl:
+                    with open(path, "r", encoding='utf-8') as lbl:
                         for line in lbl:
                             if "<start_date_time>" in line:
                                 start_time = line.split("<start_date_time>")[-1].split(
@@ -544,7 +544,7 @@ class ChecksumProduct(Product):
             #
             # Write the checksum file.
             #
-            with open(self.path, "w") as c:
+            with open(self.path, "w", encoding='utf-8') as c:
                 for entry in md5_list:
                     entry = add_carriage_return(entry, self.setup.eol, self.setup)
                     c.write(entry)

--- a/src/pds/naif_pds4_bundler/classes/product/product_inventory.py
+++ b/src/pds/naif_pds4_bundler/classes/product/product_inventory.py
@@ -161,9 +161,9 @@ class InventoryProduct(Product):
         # If there is an existing version we need to add the items from
         # the previous version as SECONDARY members
         #
-        with open(self.path, "w+") as f:
+        with open(self.path, "w+", encoding='utf-8') as f:
             if self.path_current:
-                with open(self.path_current, "r") as r:
+                with open(self.path_current, "r", encoding='utf-8') as r:
                     for line in r:
                         if "P,urn" in line:
                             #
@@ -205,7 +205,7 @@ class InventoryProduct(Product):
                 f"{self.setup.bundle_directory}/{self.setup.volume_id}/index/index.tab"
             )
 
-            with open(existing_index, "r") as f:
+            with open(existing_index, "r", encoding='utf-8') as f:
                 for line in f:
                     if line.strip() != "":
                         index_row = line.split(",")
@@ -260,7 +260,7 @@ class InventoryProduct(Product):
         file_types = []
 
         line_for_length = ""
-        with open(self.path, "w+") as f:
+        with open(self.path, "w+", encoding='utf-8') as f:
             for row in index:
                 rows += 1
                 line = ""
@@ -337,7 +337,7 @@ class InventoryProduct(Product):
         for product in self.collection.product:
             if type(product).__name__ != "InventoryProduct":
                 product_found = False
-                with open(self.path, "r") as c:
+                with open(self.path, "r", encoding='utf-8') as c:
                     for line in c:
                         if product.lid in line:
                             product_found = True

--- a/src/pds/naif_pds4_bundler/classes/product/product_kernel.py
+++ b/src/pds/naif_pds4_bundler/classes/product/product_kernel.py
@@ -201,7 +201,7 @@ class SpiceKernelProduct(Product):
         get_token = False
         description = False
 
-        with open(kernel_list_file, "r") as lst:
+        with open(kernel_list_file, "r", encoding='utf-8') as lst:
             for line in lst:
                 if self.name in line:
                     get_token = True
@@ -235,7 +235,7 @@ class SpiceKernelProduct(Product):
         get_token = False
         maklabel_options = False
 
-        with open(kernel_list_file, "r") as lst:
+        with open(kernel_list_file, "r", encoding='utf-8') as lst:
             for line in lst:
                 if self.name in line:
                     get_token = True
@@ -261,7 +261,7 @@ class SpiceKernelProduct(Product):
         coverage = []
         product_label = self.path.split(".")[0] + ".xml"
         if os.path.exists(product_label):
-            with open(product_label, "r") as lbl:
+            with open(product_label, "r", encoding='utf-8') as lbl:
                 start = ""
                 stop = ""
                 for line in lbl:
@@ -340,7 +340,7 @@ class SpiceKernelProduct(Product):
         :return: List of IDs present in the IK
         :rtype: list
         """
-        with open(f"{self.path}", "r") as f:
+        with open(f"{self.path}", "r", encoding='utf-8') as f:
 
             id_list = []
             parse_bool = False

--- a/src/pds/naif_pds4_bundler/classes/product/product_metakernel.py
+++ b/src/pds/naif_pds4_bundler/classes/product/product_metakernel.py
@@ -704,8 +704,8 @@ class MetaKernelProduct(Product):
 
         metakernel_dictionary = vars(self)
 
-        with open(self.path, "w+") as f:
-            with open(self.template, "r") as t:
+        with open(self.path, "w+", encoding='utf-8') as f:
+            with open(self.template, "r", encoding='utf-8') as t:
                 for line in t:
                     line = line.rstrip()
                     for key, value in metakernel_dictionary.items():

--- a/src/pds/naif_pds4_bundler/classes/product/product_orbnum.py
+++ b/src/pds/naif_pds4_bundler/classes/product/product_orbnum.py
@@ -234,7 +234,7 @@ class OrbnumFileProduct(Product):
         :return: ORBNUM header line
         """
         header = []
-        with open(self.path, "r") as o:
+        with open(self.path, "r", encoding='utf-8') as o:
 
             if int(self._orbnum_type["header_start_line"]) > 1:
                 for _i in range(int(self._orbnum_type["header_start_line"]) - 1):
@@ -273,7 +273,7 @@ class OrbnumFileProduct(Product):
         :return: OrbNum file header length
         """
         header_length = 0
-        with open(self.path, "r", newline="") as o:
+        with open(self.path, "r", encoding='utf-8', newline="") as o:
             lines = 0
             header_start = int(self._orbnum_type["header_start_line"])
             for line in o:
@@ -295,7 +295,7 @@ class OrbnumFileProduct(Product):
         :return: sample record line
         """
         sample_record = ""
-        with open(self.path, "r") as o:
+        with open(self.path, "r", encoding='utf-8') as o:
             lines = 0
             header_start = int(self._orbnum_type["header_start_line"])
             for line in o:
@@ -357,7 +357,7 @@ class OrbnumFileProduct(Product):
         """
         blank_records = []
 
-        with open(self.path, "r") as o:
+        with open(self.path, "r", encoding='utf-8') as o:
             previous_orbit_number = None
             records = 0
             lines = 0
@@ -460,8 +460,8 @@ class OrbnumFileProduct(Product):
             # Following the name, write the new file and remove the
             # provided orbnum file.
             #
-            with open(self.path, "r") as o:
-                with open(path, "w") as n:
+            with open(self.path, "r", encoding='utf-8') as o:
+                with open(path, "w", encoding='utf-8') as n:
                     i = 1
                     for line in o:
                         if i > int(header_start) + 1:

--- a/src/pds/naif_pds4_bundler/classes/product/product_readme.py
+++ b/src/pds/naif_pds4_bundler/classes/product/product_readme.py
@@ -75,9 +75,9 @@ class ReadmeProduct(Product):
             else:
                 handle_npb_error("Readme file provided via configuration does not exist.")
         elif not os.path.isfile(self.path):
-            with open(self.path, "w+") as f:
+            with open(self.path, "w+", encoding='utf-8') as f:
                 with open(
-                    self.setup.templates_directory + "/template_readme.txt", "r"
+                    self.setup.templates_directory + "/template_readme.txt", "r", encoding='utf-8'
                 ) as t:
                     for line in t:
                         if "$SPICE_NAME" in line:

--- a/src/pds/naif_pds4_bundler/classes/product/product_spiceds.py
+++ b/src/pds/naif_pds4_bundler/classes/product/product_spiceds.py
@@ -148,8 +148,8 @@ class SpicedsProduct(Product):
         time_string = today.strftime("%Y-%m-%dT%H:%M:%S.%f")
         temporary_file = f"{self.path}.{time_string}"
 
-        with open(self.path, "r") as s:
-            with open(temporary_file, "w+") as t:
+        with open(self.path, "r", encoding='utf-8') as s:
+            with open(temporary_file, "w+", encoding='utf-8') as t:
                 for line in s:
                     line = add_carriage_return(line, self.setup.eol_pds4, self.setup)
                     t.write(line)
@@ -177,9 +177,9 @@ class SpicedsProduct(Product):
         #
         generate_spiceds = True
         if self.latest_spiceds:
-            with open(self.path) as f:
+            with open(self.path, encoding='utf-8') as f:
                 spiceds_current = f.readlines()
-            with open(self.latest_spiceds) as f:
+            with open(self.latest_spiceds, encoding='utf-8') as f:
                 spiceds_latest = f.readlines()
 
             differ = difflib.Differ(charjunk=difflib.IS_CHARACTER_JUNK)

--- a/src/pds/naif_pds4_bundler/classes/setup.py
+++ b/src/pds/naif_pds4_bundler/classes/setup.py
@@ -65,7 +65,7 @@ class Setup:
         # Converting XML setup file into a dictionary and then into
         # attributes for the object.
         #
-        config = Path(args.config).read_text()
+        config = Path(args.config).read_text(encoding='utf-8')
         entries = etree_to_dict(ET.XML(config))
 
         #
@@ -745,7 +745,7 @@ class Setup:
             try:
                 xml_tag = "<Identification_Area>"
                 with open(
-                    self.templates_directory + os.sep + "template_bundle.xml", "r"
+                    self.templates_directory + os.sep + "template_bundle.xml", "r", encoding='utf-8'
                 ) as t:
                     for line in t:
                         if xml_tag in line:
@@ -1150,7 +1150,7 @@ class Setup:
                 + os.sep
                 + f"{self.mission_acronym}_{self.run_type}_"
                 f"{int(self.release):02d}.file_list",
-                "w",
+                "w", encoding='utf-8'
             ) as l:
                 for file in self.file_list:
                     l.write(file + "\n")
@@ -1164,7 +1164,7 @@ class Setup:
                 + os.sep
                 + f"{self.mission_acronym}_{self.run_type}_"
                 f"{int(self.release):02d}.checksum",
-                "w",
+                "w", encoding='utf-8'
             ) as l:
                 for element in self.checksum_registry:
                     l.write(element + "\n")
@@ -1223,7 +1223,7 @@ class Setup:
 
         filename = f"{self.mission_acronym}_{self.run_type}_{int(self.release):02d}"
 
-        with open(f"{self.working_directory}/{filename}.validate_config", "w") as l:
+        with open(f"{self.working_directory}/{filename}.validate_config", "w", encoding='utf-8') as l:
             l.write(
                 "# Run the PDS validate tool where the NPB working directory resides:\n"
             )

--- a/src/pds/naif_pds4_bundler/pipeline/runtime.py
+++ b/src/pds/naif_pds4_bundler/pipeline/runtime.py
@@ -24,7 +24,7 @@ def clear_run(setup: Setup) -> None:
     if os.path.isfile(setup.args.clear):
 
         # Read the kernel list file provided in the "clear" argument.
-        with open(setup.args.clear, "r") as handle:
+        with open(setup.args.clear, "r", encoding='utf-8') as handle:
             lines = handle.readlines()
 
         # Remove files from the staging area.

--- a/src/pds/naif_pds4_bundler/utils/files.py
+++ b/src/pds/naif_pds4_bundler/utils/files.py
@@ -242,8 +242,8 @@ def add_crs_to_file(file, eol, setup=False):
     """
     try:
         file_crs = file.split(".")[0] + "crs_tmp"
-        with open(file, "r") as r:
-            with open(file_crs, "w+") as f:
+        with open(file, "r", encoding='utf-8') as r:
+            with open(file_crs, "w+", encoding='utf-8') as f:
                 for line in r:
                     line = add_carriage_return(line, eol, setup)
                     f.write(line)
@@ -278,9 +278,9 @@ def fill_template(list_object, product_file, product_dictionary):
     :param product_dictionary: Dictionary of keys to replace
     :type product_dictionary: dict
     """
-    with open(product_file, "w") as f:
+    with open(product_file, "w", encoding='utf-8') as f:
 
-        with open(list_object.template, "r") as t:
+        with open(list_object.template, "r", encoding='utf-8') as t:
             for line in t:
                 line = line.rstrip()
                 for key, value in product_dictionary.items():
@@ -309,7 +309,7 @@ def get_context_products(setup):
     registered_context_products_file = str(
         files('pds.naif_pds4_bundler.data').joinpath('registered_context_products.json')
     )
-    with open(registered_context_products_file, "rt") as f:
+    with open(registered_context_products_file, "rt", encoding='utf-8') as f:
         context_products = json.load(f)["Product_Context"]
 
     #
@@ -384,7 +384,7 @@ def mk_to_list(mk, setup):
     path_symbol = ""
     get_symbol = False
     ker_mk_list = []
-    with open(mk, "r") as f:
+    with open(mk, "r", encoding='utf-8') as f:
         for line in f:
             if path_symbol:
                 if path_symbol in line:
@@ -478,7 +478,7 @@ def get_latest_kernel(
 
     if mks:
         for mk in mks:
-            with open(mk, "r") as m:
+            with open(mk, "r", encoding='utf-8') as m:
                 for line in m:
                     if re.findall(pattern, line):
                         kernels_with_path.append(line.strip())
@@ -576,9 +576,9 @@ def compare_files(fromfile, tofile, dest_dir, display):
     :return: True if the files are different, False if they are the same.
     :rtype: bool
     """
-    with open(fromfile) as ff:
+    with open(fromfile, encoding='utf-8') as ff:
         fromlines = ff.readlines()
-    with open(tofile) as tf:
+    with open(tofile, encoding='utf-8') as tf:
         tolines = tf.readlines()
 
     if fromlines == tolines:
@@ -607,7 +607,7 @@ def compare_files(fromfile, tofile, dest_dir, display):
             f"{fromfile.split(os.sep)[-1].replace('.', '_')}_"
             f"{tofile.split(os.sep)[-1].replace('.', '_')}"
             f".html",
-            "w",
+            "w", encoding='utf-8'
         )
         diff_html.writelines(diff)
         diff_html.close()
@@ -735,7 +735,7 @@ def checksum_from_registry(path, working_directory):
 
     for checksum_registry in checksum_registries:
         if not checksum_found:
-            with open(checksum_registry, "r") as lbl:
+            with open(checksum_registry, "r", encoding='utf-8') as lbl:
                 for line in lbl:
                     if path in line:
                         checksum = line.split()[-1]
@@ -760,7 +760,7 @@ def checksum_from_label(path):
     checksum = ""
     product_label = path.split(".")[0] + ".xml"
     if os.path.exists(product_label):
-        with open(product_label, "r") as lbl:
+        with open(product_label, "r", encoding='utf-8') as lbl:
             for line in lbl:
                 if "<md5_checksum>" in line:
                     checksum = line.split("<md5_checksum>")[-1]
@@ -830,7 +830,7 @@ def string_in_file(file, str_to_check, repetitions=1):
     :rtype: bool
     """
     lines_with_string = 0
-    with open(file, "r") as r:
+    with open(file, "r", encoding='utf-8') as r:
         for line in r:
             if str_to_check in line:
                 lines_with_string += 1
@@ -852,7 +852,7 @@ def replace_string_in_file(file, old_string, new_string, setup):
     :type new_string: str
     :param setup: NPB run Setup
     """
-    reading_file = open(file, "r")
+    reading_file = open(file, "r", encoding='utf-8')
 
     new_file_content = ""
     for line in reading_file:
@@ -860,7 +860,7 @@ def replace_string_in_file(file, old_string, new_string, setup):
         new_file_content += add_carriage_return(new_line, setup.eol_pds3, setup)
     reading_file.close()
 
-    writing_file = open("temp.file", "w")
+    writing_file = open("temp.file", "w", encoding='utf-8')
     writing_file.write(new_file_content)
     writing_file.close()
 
@@ -902,7 +902,7 @@ def product_mapping(name, setup, cleanup=True):
     get_map = False
     mapping = False
 
-    with open(kernel_list_file, "r") as lst:
+    with open(kernel_list_file, "r", encoding='utf-8') as lst:
         for line in lst:
             if name in line:
                 get_map = True
@@ -1066,7 +1066,7 @@ def check_badchar(file):
     """
     error = []
     line_num = 1
-    with open(file, "r") as f:
+    with open(file, "r", encoding='utf-8') as f:
         for line in f:
             if not line.isascii():
                 char_count = 0
@@ -1123,7 +1123,7 @@ def check_line_length(file):
     """
     error = []
     line_num = 1
-    with open(file, "r") as f:
+    with open(file, "r", encoding='utf-8') as f:
         for line in f:
             if len(line) > 80:
                 error.append(f"Line {line_num} is longer than 80 characters")

--- a/src/pds/naif_pds4_bundler/utils/time.py
+++ b/src/pds/naif_pds4_bundler/utils/time.py
@@ -389,7 +389,7 @@ def pds3_label_gen_date(file):
     """
     generation_date = "N/A"
 
-    with open(file, "r") as f:
+    with open(file, "r", encoding='utf-8') as f:
 
         for line in f:
             if "PRODUCT_CREATION_TIME" in line:

